src/main/java/org/broadinstitute/hellbender/engine/TwoPassVariantWalker.java

```diff
@@ -0,0 +1,70 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.VariantFilter;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.util.stream.StreamSupport;
+
+public abstract class TwoPassVariantWalker extends VariantWalker {
+
+    @Override
+    public void traverse(){
+        final VariantFilter variantFilter = makeVariantFilter();
+        final CountingReadFilter readFilter = makeReadFilter();
+
+        // First pass through the variant
+        traverseVariants(variantFilter, readFilter, this::firstPassApply);
+        logger.info("Finished first pass through the variants");
+
+        afterFirstPass();
+
+        // Second pass
+        logger.info("Starting second pass through the variants");
+        traverseVariants(variantFilter, readFilter, this::secondPassApply);
+        logger.info(readFilter.getSummaryLine());
```

Comment:
This  accumulates the filter count across both passes and displays the aggregate total. The TwoPassReadWalker does the same thing, though I wonder if thats a feature, given that the second pass is usually an implementation detail.

Reply:
I talked to Louis and James a bit about this. I don't think we need two different `VariantFilters`, so I'll just call `makeVariantFilter()` twice,  in the hope that calling it the second time resets the counter